src/app/payment/paymentMethod/StripeUPEPaymentMethod.tsx

```diff
@@ -0,0 +1,66 @@
+import { PaymentInitializeOptions } from '@bigcommerce/checkout-sdk';
+import React, { useCallback, FunctionComponent } from 'react';
+
+import { withCheckout, CheckoutContextProps } from '../../checkout';
+import { TranslatedString } from '../../locale';
+import { withHostedCreditCardFieldset, WithInjectedHostedCreditCardFieldsetProps } from '../hostedCreditCard';
+
+import HostedWidgetPaymentMethod, { HostedWidgetPaymentMethodProps } from './HostedWidgetPaymentMethod';
+
+export type StripePaymentMethodProps = Omit<HostedWidgetPaymentMethodProps, 'containerId'>;
+
+interface WithCheckoutStripePaymentMethodProps {
+    storeUrl: string;
+}
+export enum StripeElementType {
+    card = 'card',
+}
+const StripeUPEPaymentMethod: FunctionComponent<StripePaymentMethodProps & WithInjectedHostedCreditCardFieldsetProps & WithCheckoutStripePaymentMethodProps> = ({
+    initializePayment,
+    getHostedFormOptions,
+    getHostedStoredCardValidationFieldset,
+    hostedStoredCardValidationSchema,
+    method,
+    storeUrl,
+    ...rest
+    }) => {
+    const paymentMethodType = method.id as StripeElementType;
+    const containerId = `stripe-${paymentMethodType}-component-field`;
+
+    const initializeStripePayment: HostedWidgetPaymentMethodProps['initializePayment'] = useCallback(async (options: PaymentInitializeOptions) => {
+        return initializePayment({ ...options,
+            stripeupe: { containerId}});
+    }, [initializePayment, containerId]);
+
+    return <>
+        <HostedWidgetPaymentMethod
+            { ...rest }
+            containerId={ containerId }
+            hideContentWhenSignedOut
+            initializePayment={ initializeStripePayment }
+            method={ method }
+        />
+        {
+            method.id === 'iban' &&
```

Comment:
This is a bit weird, we say in line 27 and 16 that `method.id` will always be card, then here, we compare with `iban`.